frontend/app/src/components/ui/loading.tsx

```diff
@@ -0,0 +1,15 @@
+import { Icons } from '@/components/ui/icons.tsx'
+
+export function Spinner() {
+  return (
+    <Icons.spinner className="mr-2 h-4 w-4 animate-spin"/>
+  )
+}
+
+export function Loading() {
+  return (
+    <div className="flex flex-row flex-1 w-full h-full">
+      <Loading/>
```

Comment:
Doesn't this call itself?

Reply:
whoops, good catch, I messed up my last refactor 😅